### PR TITLE
 Document `GET /source/{project_name}/{package_name}` with OpenAPI

### PR DESCRIPTION
**Pull request by eduardoj**
### For reviewers

Access to the documented enpoint in the review-app through [this link](https://obs-reviewlab.opensuse.org/eduardoj-apidocs_newsource_package_get/apidocs-new/index.html#/Sources%20-%20Files/get_source__project_name___package_name_).

### CODE REVIEW

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml [hunk 1]
```diff
@@ -1,3 +1,352 @@
...
+    # - in: query
+    #   name: emptylink
+    #   schema:
+    #     type: string
+    #     enum:
+    #       - 1
+    #       - 0
+    #   default: 0
+    #   description:
+    #   example: 1
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_project_name_package_name.yaml:28]: I will leave documenting the `emptylink` parameter as a TODO until I figure out what is it for.

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml [hunk 2]
```diff
@@ -1,3 +1,352 @@
...
+          - issues
+          - products
+          - productrepositories
+      description: |
+        Specify which information about a package should be returned.
+
+        * `cpio`: Stream all package files as cpio.
+        * `getmultibuild`: List multibuild packages.
+        * `info`: Show source version, md5sums and build description files, among other information.
+           See this [other endpoint](#/TODO) for details.
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_project_name_package_name.yaml:98]: This endpoint with `view=info` will be documented in other entry. Once we document it, we should link from here to the new documented endpoint.

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml [hunk 3]
```diff
@@ -1,3 +1,352 @@
...
+          - 1
+          - 0
+      default: 0
+      description: Expand links.
+      example: 1
+    - in: query
+      name: extension
+      schema:
+        type: string
+      description: List sources files with the given extension.
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name.yaml:43]:
```suggestion
      description: List source files with the given extension.
```

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml [hunk 4]
```diff
@@ -1,3 +1,352 @@
...
+            $ref: '../components/schemas/api_response.yaml'
+          examples:
+            Unknown Project:
+              value:
+                code: unknown_project
+                summary: "Project not found: home:some_project"
+            Unknown Package:
+              value:
+                code: unknown_package
+                summary: "Package not found: home:some_project/some_package"
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name.yaml:346]: You could use a reference here, right?
```
  '404':
      $ref: '../components/responses/unknown_project_or_package.yaml'
```

**Review by saraycp — CHANGES_REQUESTED** [review]

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml (hunk 3 above)
Comment (eduardoj) [reply src/api/public/apidocs-new/paths/source_project_name_package_name.yaml:43]: Fixed.

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml [hunk 5]
```diff
@@ -1,3 +1,352 @@
...
+      example: txt
+    - in: query
+      name: lastworking
+      schema:
+        type: string
+        enum:
+          - 1
+          - 0
+      default: 0
+      description: Show sources of last mergeable sources in case of conflicting changes.
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/public/apidocs-new/paths/source_project_name_package_name.yaml:53]: Fixed, thanks.

File: src/api/public/apidocs-new/paths/source_project_name_package_name.yaml (hunk 4 above)
Comment (eduardoj) [reply src/api/public/apidocs-new/paths/source_project_name_package_name.yaml]: Also applied, thanks! Please take a look again.

**Review by saraycp — APPROVED** [review]

**Pull request merged**